app/assets/javascripts/float-kit/addon/services/menu.js

```diff
@@ -0,0 +1,93 @@
+import Service from "@ember/service";
+import { getOwner } from "@ember/application";
+import { action } from "@ember/object";
+import DMenuInstance from "float-kit/lib/d-menu-instance";
+import { MENU } from "float-kit/lib/constants";
+import { guidFor } from "@ember/object/internals";
+import { tracked } from "@glimmer/tracking";
+import { next } from "@ember/runloop";
+
+export default class Menu extends Service {
+  @tracked activeMenu;
+
+  get portalOutletElement() {
+    return document.querySelector(`#${MENU.portalOutletId}`);
+  }
+
+  /**
+   * Render a menu
+   *
+   * @param {Element} trigger - the element that triggered the menu, can also be an object implementing `getBoundingClientRect`
+   * @param {Object} [options] - options
+   * @param {String | Element | Component} [options.content] - Specifies the content of the menu
+   * @param {Integer} [options.maxWidth] - Specifies the maximum width of the content
+   * @param {Object} [options.model] - An object which will be passed as the `@model` argument when content is a `Component`
+   * @param {Boolean} [options.arrow] - Determines if the menu has an arrow
+   * @param {Boolean} [options.offset] - Displaces the content from its reference trigger in pixels
+   * @param {String} [options.identifier] - Add a data-identifier attribute to the trigger and the content
+   * @param {Boolean} [options.inline] - Improves positioning for trigger that spans over multiple lines
+   *
+   * @returns {Promise} An promise resolving to an instance of the menu
+   */
+  @action
+  async show(trigger, options = {}) {
+    const id = trigger.id?.length ? trigger.id : guidFor(trigger);
+    const alreadyOpened =
+      this.activeMenu && this.activeMenu.id === id && this.activeMenu.expanded;
+
+    if (this.activeMenu) {
+      this.activeMenu?.close();
+      this.activeMenu = null;
+    }
+
+    if (alreadyOpened) {
+      return;
+    }
+
+    await new Promise((resolve) => {
+      next(() => resolve());
+    });
```

Comment:
What purpose does this serve? I notice that we don't have this in `.replace`, which is surprising 🤔

Reply:
Yes... i think I don't need this anymore.